Live visualization

emaximili opened the pull request:
fixes #10

emaximili committed:
Added possibility to import / export files. 
- simple button at the bottom. 
- converted all the old examples to new input files.
- fixed visualization of inluence line bug.

emaximili committed:
removed old input files from git

emaximili committed:
Adjusted loading and saving buttons. Made info button blue. Changed beam load names.
Fixed static condensation logic:
- table on off works
- Multiple nodes now supported
- Results shown in workspace

emaximili committed:
Added error VAlidation upon switching panels:
- Separate functions for each panel
- Same checks as in ValidateAnalysisModel but adjusted for different input type

Put setup file in root directory

emaximili committed:
WIP: partial progress on issue #10

emaximili committed:
Mbeautified drawOriginalFig

emaximili committed:
WIP: On the fly visualization

emaximili committed:
WIP: Added new functions to repo

emaximili committed:
Added functionality to delete all table input. Improved the error handling.

emaximili committed:
added icons to git

emaximili committed:
merged GUI version with IO_files

emaximili committed:
Merge branch 'IO_files' into live_visualization

# Conflicts:
#	src/DSM/Zeichnen/drawOriginalFig.m
#	src/GUI/GUI.mlapp

added icons as well

emaximili committed:
WIP: Live visualization

emaximili committed:
WIP visualization: 
- Consistent s^force scaling
- Live visualization works now for most edge cases
- Scaling of axis fixed
- Improved position of labels
- visually improved angle of support when multiple beams meet at a support
- clean error handling when deleting rows with dependencies

emaximili committed:
added getTextPos functions to git

federicobetti-ethz reviewed (commented):
All good for the code you added here, but some of the boxes in #10 are not checked yet which means there is more work to do I assume.

Moreover, there are conflicts with master, as you can see in the PR here. This means that the PR is not yet really ready for review. Please proceed to fix these things first.

emaximili committed:
WIP: Live visualization

emaximili committed:
WIP

emaximili committed:
Fixed influence line visualization bug

emaximili committed:
Live visualization:
- Improved text / label location for forces and nodes. 
- Improved visualization and scaling of forces
- Added legend with option to hide things
- Systemviewer scaling improvements

emaximili committed:
added legend icon to git

emaximili committed:
Live visualization:
- Improved error hanfling edge case with proper viusalization
- Grouped beam elements for toggling in legend
- Live visualization of hinges and stiff beams now works
- refactored part o inputumwandeln for shared use

emaximili committed:
WIP

emaximili committed:
fixed bug

emaximili committed:
Stable Version:
- Imporved scaling once again
- Added Unit examples to infopanel
- fixed legend items resetting upon new visualization

emaximili committed:
removed buildStabFromInput

emaximili committed:
small improvements, cleaning up

emaximili committed:
Fixed Scaling, Fixed visual bug of moment lines not connecting (numerical artifact), Improved force diagram style, added red dots for maxima, minima and zero-points.

emaximili committed:
Adjusted colors of forces:
- all external loads are red
- reaction forces are green
- internal forces are blue (force-diagram)
- node markers are pink
- beam markers are orange

Added Numbering to beams with triangle marking the direction, Adjusted naming convention of Nodalloads and memeber loads.

emaximili commented:
Hi Federico,
I’ve updated everything according to what we discussed in our meeting on Monday. From my side, the branch should now be ready to merge. It would be great if you could have a look and give me the green light.
I think I need to resolve the conflicts during merging (/git merge origin/master). Or do you have another idea how to resolve the conflicts?
Best, Max

federicobetti-ethz commented:
Hi Max, you need to resolve the conflicts before merging this into master. Please make sure the branch master is up to date, then come back on this branch and do git merge master. Push when you resolved all the conflicts and I shall take a look.

emaximili committed:
resolved merge conflicts

emaximili committed:
updated legend color of distr. loads

emaximili commented:
I solved the merge conflicts now. Thanks for the instructions!
